File: services/parser/testdata/lang/python/repo.py
import functools
import os.path as osp
from .util import wrap, unwrap as peel


def trace(fn):
    @functools.wraps(fn)
    def inner(*args):
        return fn(*args)

    return inner


class Repo:
    def __init__(self, ident):
        self.ident = ident

    @trace
    def sync(self):
        label = self.label()
        return wrap(f"{describe(label)}")

    def label(self):
        return peel(self.ident)

    class Nested:
        def deep(self):
            return describe("nested")


async def fetch(repo):
    return await repo.sync()


def describe(value):
    return str(value)


def apply(values):
    return [render(v) for v in values if keep(v)]


def render(value):
    return osp.basename(str(value))


def keep(value):
    return bool(value)

File: services/parser/testdata/lang/rust/repo.rs
use std::fmt::Write as FmtWrite;
use crate::util::{wrap, unwrap as peel};

pub struct Repo {
    id: String,
}

impl Repo {
    pub fn sync(&self) -> String {
        let label = self.label();
        println!("{}", describe(&label));
        wrap(label)
    }

    fn label(&self) -> String {
        peel(self.id.clone())
    }
}

pub fn describe(value: &str) -> String {
    value.to_owned()
}

pub fn apply(values: Vec<i32>) -> Vec<String> {
    values.iter().map(|v| render(*v)).collect()
}

fn render(value: i32) -> String {
    match value {
        0 => empty(),
        _ => describe(&value.to_string()),
    }
}

fn empty() -> String {
    String::new()
}
